File: java/ContainerWithMostWater.java
/*
 * Filename: ContainerWithMostWater.java
 * Author: Jun Xu
 * Email: junx@bu.edu
 * Created Time: Wed 11 Mar 2015 01:42:11 PM EDT
 */

public class Solution {
	public int maxArea(int[] height) {
		int i = 0;
		int j = height.length - 1;
		int max = 0;

		while (i < j) {
			max = Math.max(max, (j - i) * Math.min(height[i], height[j]));
			if (height[i] < height[j]) {
				i++;
			} else {
				j--;
			}
		}

		return max;
	}
}

File: cpp/container_with_most_water.cc
/*
 * Filename: container_with_most_water.cc
 * Author: Jun Xu
 * Email: jxu.vimacs@gmail.com
 * Created Time: Thu 23 Jul 2015 08:38:25 PM EDT
 */

#include <iostream>
using namespace std;

class Solution {
	public:
		int maxArea(vector<int>& height) {
			int i = 0;
			int j = height.size() - 1;
			int res = 0;

			while (i < j) {
				if (height[i] > height[j]) {
					res = max(res, (j-i) * height[j]);
					j--;
				} else {
					res = max(res, (j-i) * height[i]);
					i++;
				}
			}

			return res;
		}
};
